File: warm_up/week_05/day_04/cipher.js
var Plain = "abcdefghijklmnopqrstuvwxyz";
var Cipher = Plain.split('').reverse().join('');

var atbash = function(string) {
  var solution = '';
  for ( var i =0; i < string.length; i++ ) {
    var cipher_index = Plain.indexOf(string[i]);
    solution += Cipher[cipher_index]
  }
  console.log(solution);
};

console.log(Plain);
console.log(Cipher);

atbash('test');
atbash('gvhg');

File: warm_up/week_05/day_04/cipher.rb
# # Atbash Cipher
#
# The Atbash cipher is a simple substitution cipher that relies on transposing all the letters in the alphabet such that the resulting alphabet is backwards. The first letter is replaced with the last letter, the second with the
# second-last, and so on.
#
# An Atbash cipher for the Latin alphabet would be as follows:
#
# ```plain
# Plain:  abcdefghijklmnopqrstuvwxyz
# Cipher: zyxwvutsrqponmlkjihgfedcba
# ```
#
# It is a very weak cipher because it only has one possible key, and it is a
# simple monoalphabetic substitution cipher. However, this may not have been an issue in the cipher's time.
#
# ## Examples
# - Encoding "test" gives "gvhg"
# - Decoding "gvhg" gives "test"

def atbash(string)
  plain = ('a'..'z').to_a
  cipher = plain.reverse
  arr = string.split('')
  solution = []

  # Ruby helper to iterate through string:
  # string.each_char do |el|

  arr.each do |el|
    cipher_index = plain.index(el)
    solution.push(cipher[cipher_index])
  end

  p solution.join('')
end

atbash('test')
atbash('gvhg')
